### DISCUSSION TITLE
Copilot Monthly Digest - June edition 🛠️

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#129577] [Copilot News and Announcements]
👋  This month, we've introduced exciting updates to GitHub Copilot! You can now build extensions through our new Copilot Extensions guide. We've added a centralized enterprise endpoint for listing Copilot seats, and Copilot Chat in both Visual Studio Code and Visual Studio has been enhanced with the ability to access knowledge bases and search Bing for contextual information. Additionally, Copilot Chat now provides enriched answers from across your entire codebase, not just open tabs. We are also thrilled to announce the availability of compliance reports for GitHub Copilot Business and Copilot Enterprise, including a SOC 2 Type I report and inclusion in our ISO 27001 certificate. These updates are designed to streamline your development process and enhance your coding experience with GitHub Copilot

## Copilot Extensions: Everything You Need to Know to Get Started 📚

We have published a new guide to Copilot Extensions within the [Copilot Partner Program (CPP)](https://github.com/features/preview/copilot-partner-program) on our partners page. To currently build extensions, you need to be part of the CPP. Click [here](https://partner.github.com/2024/06/18/copilot-extensions.html) to read the full article and learn more about the program.

## Centralized Enterprise Endpoint for Copilot Seats 🌐

With today’s update, we’ve added a centralized enterprise endpoint for listing Copilot seats and associated metadata across the enterprise. We have also updated the necessary scope to `read:enterprise`. Now, all enterprise admins can quickly gather their enterprise’s Copilot seats details with just one API request!

## Compliance Reports Now Available 📋
We're thrilled to announce that compliance reports are now available for GitHub Copilot Business and Copilot Enterprise! GitHub has published a SOC 2 Type I report for Copilot Business, showcasing the necessary controls for service security. Both Copilot Business and Enterprise are now included in GitHub’s ISO 27001 certification, reflecting our high standards for security and compliance. We’ll also be part of the SOC 2 Type II report coming late 2024. Learn more about accessing compliance reports and certifications in our [documentation](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/overview/enabling-github-copilot-enterprise-features).

----

## Enhanced Copilot Chat in Visual Studio Code 📝

[GitHub Copilot Enterprise](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/overview/about-github-copilot-enterprise) users in Visual Studio Code can now enjoy several new features:

### Access Your Copilot Knowledge Bases 📖

You can now access your [knowledge bases](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/copilot-chat-in-github/managing-copilot-knowledge-bases) from any Copilot Chat conversation by typing `@github #kb`, selecting a knowledge base from the list, and then entering your question. Copilot will respond, using the Markdown documentation in your knowledge base as context for its answer.

-  Search with the Context of the Web 🔍: GitHub Copilot can now search Bing within chat conversations to answer questions and find information outside of its general knowledge or your codebase. Start your message with `@github` and Copilot will decide when to use Bing, or you can add `#web` to your query. For example, you can ask, `@github What is the latest LTS version of Node.js?`.

Bing search is only available if enabled by an administrator. For more details, see [“Enabling GitHub Copilot Enterprise features”](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/overview/enabling-github-copilot-enterprise-features).

----

## Enhanced Copilot Chat in Visual Studio 🖥️

[GitHub Copilot Enterprise](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/overview/about-github-copilot-enterprise) subscribers in Visual Studio can now use Copilot Chat to get answers enriched with context from their entire repository (not just open tabs!) and Bing search results. To try out this functionality, install [Visual Studio 17.11 Preview 2](https://visualstudio.microsoft.com/vs/preview/), and then check out the [docs](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-chat/copilot-chat-in-ides/about-github-copilot-chat-in-your-ide?tool=visualstudio).

- Search with the Context of the Web 🌍: GitHub Copilot can now search Bing to find information outside of its general knowledge or your codebase. When you mention `@github`, Copilot will intelligently decide when to use Bing. You can ask questions like `@github What are the breaking changes in Next.js v14?`.

----

## Mastering Copilot: Top Learning Resources 🔖 

The GitHub blog features several insightful articles to help you get the most out of GitHub Copilot. Here are some highlights:

-  **[10 Unexpected Ways to Use GitHub Copilot](https://github.blog/2023-06-07-10-unexpected-ways-to-use-github-copilot/)**: This blog post explores innovative uses of Copilot, from generating documentation and fixing code inline to explaining error messages and debugging GitHub Actions workflows. It demonstrates how Copilot extends beyond basic code completion to become an integral part of your development toolkit.
- **[Introducing GitHub Copilot Extensions](https://github.blog/2023-05-31-introducing-github-copilot-extensions/)**: Learn about the new Copilot Extensions and how they can streamline your development workflow. The article explains how extensions from partners like Sentry, Azure, and Docker can enhance your coding experience by allowing you to resolve issues, retrieve data, and deploy applications directly from your IDE.

- **[Exploring Copilot for Azure](https://github.blog/2023-06-08-exploring-copilot-for-azure/)**: This post showcases how GitHub Copilot integrates with Azure, helping developers with everything from selecting services to deploying applications. It emphasizes the seamless natural language interface that allows for intuitive and efficient development on the Azure platform.

- **[Fixing Errors and Generating Documentation](https://github.blog/2023-06-14-fixing-errors-and-generating-documentation-with-copilot/)**: Discover how Copilot can assist in generating language-specific documentation and provide detailed explanations for error messages in your terminal, enhancing your overall productivity and understanding of your code.

For a deeper dive into these topics and more, visit the [GitHub Copilot blog](https://github.blog/tag/github-copilot/) or [subscribe to  our newsletter](https://github.blog/tag/github-copilot/#:~:text=Subscribe%20to%20our%20newsletter). 🚀

**Comment by SerratedSharp** [comment]
@Akash1134 It seems like `@github` doesn't use general Bing searches to augment results the way Copilot in Edge does. Is it only for repos?  Trying to get useful responses on any leading edge technologies is difficult.  Using Copilot directly in Edge gives more current responses since it seems to augment the results with a search, but it lacks the context of my code.  

Is this something that is being considered in the roadmap?
